src/persist/src/indexed/runtime.rs

```diff
@@ -578,52 +578,67 @@ impl<U: Buffer, L: Blob> RuntimeImpl<U, L> {
-            Cmd::Write(updates, res) => {
-                let write_res = self.indexed.write_sync(updates);
-                // TODO: Move this to a Cmd::Tick or something.
-                let step_res = self.indexed.step();
-                res.fill(step_res.and_then(|_| write_res));
-            }
-            Cmd::Seal(ids, ts, res) => {
-                let seal_res = self.indexed.seal(ids, ts);
-                // TODO: Move this to a Cmd::Tick or something.
-                let step_res = self.indexed.step();
-                res.fill(step_res.and_then(|_| seal_res));
-            }
-            Cmd::AllowCompaction(id, ts, res) => {
-                let r = self.indexed.allow_compaction(id, ts);
-                res.fill(r);
-            }
-            Cmd::Snapshot(id, res) => {
-                let r = self.indexed.snapshot(id);
-                res.fill(r);
+
+        loop {
+            match self.rx.try_recv() {
+                Ok(cmd) => cmds.push(cmd),
+                Err(crossbeam_channel::TryRecvError::Empty) => {
+                    break;
+                }
+                Err(crossbeam_channel::TryRecvError::Disconnected) => {
+                    // All Runtime handles hung up. Drop should have shut things down
+                    // nicely, so this is unexpected.
+                    return false;
+                }
             }
-            Cmd::Listen(id, listen_fn, res) => {
-                let r = self.indexed.listen(id, listen_fn);
-                res.fill(r);
+        }
+
+        for cmd in cmds {
+            match cmd {
+                Cmd::Stop(res) => {
+                    res.fill(self.indexed.close());
+                    return false;
+                }
+                Cmd::Register(id, (key_codec_name, val_codec_name), res) => {
+                    let r = self.indexed.register(&id, key_codec_name, val_codec_name);
+                    res.fill(r);
+                }
+                Cmd::Destroy(id, res) => {
+                    let r = self.indexed.destroy(&id);
+                    res.fill(r);
+                }
+                Cmd::Write(updates, res) => {
+                    let r = self.indexed.write_sync(updates);
+                    res.fill(r);
+                }
+                Cmd::Seal(ids, ts, res) => {
+                    let r = self.indexed.seal(ids, ts);
+                    res.fill(r);
+                }
+                Cmd::AllowCompaction(id, ts, res) => {
+                    let r = self.indexed.allow_compaction(id, ts);
+                    res.fill(r);
+                }
+                Cmd::Snapshot(id, res) => {
+                    let r = self.indexed.snapshot(id);
+                    res.fill(r);
+                }
+                Cmd::Listen(id, listen_fn, res) => {
+                    let r = self.indexed.listen(id, listen_fn);
+                    res.fill(r);
+                }
             }
         }
+        let _ = self.indexed.step();
```

Comment:
I couldn't think of a good way to pass this response back (which command do you send it to?) or a good reason to (even if step happened to fail it doesn't really affect the success or failure of the other commands)

Reply:
yep makes sense. lets log::warn! here. it'll be spammy for now but we can fix that later

Reply:
done